src/favourites/reducer.ts

```diff
@@ -0,0 +1,108 @@
+import { createSlice } from '@reduxjs/toolkit';
+import {
+  FETCH_FAVOURITES,
+  FETCH_FAVOURITE_ATTRIBUTES,
+  RECEIVE_FAVOURITES,
+  RECEIVE_FAVOURITE_ATTRIBUTES,
+  Favourite,
+  ReceiveFavouritesAction,
+  ReceiveFavouriteAttributesAction,
+  ADD_FAVOURITE_TARGET,
+  AddFavouriteTargetAction,
+  REMOVE_FAVOURITE_TARGET,
+  RemoveFavouriteTargetAction,
+} from './types';
+import { ApiAttributes } from '../api/types';
+import { PlotSearchTarget } from '../plotSearch/types';
+
+type CurrentDisplayState = {
+  favourites: Favourite;
```

Comment:
The ReceiveFavourites action stores an array here, but the type is singular here and in all operations so far.

Reply:
I didnt do much more than refactored "favourites => favourite" at this point.